packages/editor/src/components/document-outline/index.js

```diff
@@ -121,7 +121,10 @@ export const DocumentOutline = ( {
 							isValid={ isValid }
 							isDisabled={ hasOutlineItemsDisabled }
 							href={ `#block-${ item.clientId }` }
-							onSelect={ onSelect }
+							onSelect={ () => {
+								selectBlock( item.clientId );
+								onSelect?.();
```

Comment:
This this used to work because of the link to the anchor. I wonder if we can trigger a URL change in the iframe...